### PR TITLE
[netdata] update/simplify public API documentations

### PR DESCRIPTION
**Pull request by abtink**
This commit contains no logic/behavior change and only contains
typos/style fixes and documentation updates and simplifications.

### CODE REVIEW

**Review by simonlingoogle — APPROVED** [review]
LGTM :+1:

File: include/openthread/netdata.h [hunk 1]
```diff
@@ -57,67 +57,18 @@ typedef uint32_t otNetworkDataIterator; ///< Used to iterate through Network Dat
...
-    /**
-     * The Border Agent Rloc.
-     */
-    uint16_t mRloc16;
+    otIp6Prefix mPrefix;           ///< The IPv6 prefix.
+    signed int  mPreference : 2;   ///< A 2-bit signed int preference (`OT_ROUTE_PREFERENCE_*` values).
+    bool        mPreferred : 1;    ///< Whether or not prefix is preferred.
+    bool        mSlaac : 1;        ///< Whether or not prefix can be used for address auto-configuration (SLAAC).
+    bool        mDhcp : 1;         ///< Whether or not border router is DHCPv6 Agent.
+    bool        mConfigure : 1;    ///< Whether or not DHP6 Agent supplying other config data.
```
Comment (Irving-cl) [inline include/openthread/netdata.h:65]:
```suggestion
    bool        mConfigure : 1;    ///< Whether or not DHCPv6 Agent supplying other config data.
```

**Review by Irving-cl — APPROVED** [review]
LGTM :+1:

File: include/openthread/netdata.h [hunk 2]
```diff
@@ -57,67 +57,18 @@ typedef uint32_t otNetworkDataIterator; ///< Used to iterate through Network Dat
...
-     */
-    uint16_t mRloc16;
+    otIp6Prefix mPrefix;           ///< The IPv6 prefix.
+    signed int  mPreference : 2;   ///< A 2-bit signed int preference (`OT_ROUTE_PREFERENCE_*` values).
+    bool        mPreferred : 1;    ///< Whether or not prefix is preferred.
+    bool        mSlaac : 1;        ///< Whether or not prefix can be used for address auto-configuration (SLAAC).
+    bool        mDhcp : 1;         ///< Whether or not border router is DHCPv6 Agent.
+    bool        mConfigure : 1;    ///< Whether or not DHCPv6 Agent supplying other config data.
+    bool        mDefaultRoute : 1; ///< Whether or not border router is a default router for prefix.
+    bool        mOnMesh : 1;       ///< Whether or not if this prefix is considered on-mesh.
```
Comment (jwhui) [inline include/openthread/netdata.h:67]:
```suggestion
    bool        mOnMesh : 1;       ///< Whether or not this prefix is considered on-mesh.
```

File: include/openthread/netdata.h [hunk 3]
```diff
@@ -57,67 +57,18 @@ typedef uint32_t otNetworkDataIterator; ///< Used to iterate through Network Dat
...
-     */
-    bool mDp : 1;
-
-    /**
-     * The Border Agent Rloc.
-     */
-    uint16_t mRloc16;
+    otIp6Prefix mPrefix;           ///< The IPv6 prefix.
+    signed int  mPreference : 2;   ///< A 2-bit signed int preference (`OT_ROUTE_PREFERENCE_*` values).
+    bool        mPreferred : 1;    ///< Whether or not prefix is preferred.
```
Comment (jwhui) [inline include/openthread/netdata.h:62]: nit: how about making even more concise (here and below)
```suggestion
    bool        mPreferred : 1;    ///< Whether prefix is preferred.
```

**Review by jwhui — APPROVED** [review]

**Pull request merged**